File: src/NBdbm/Classes/Fachada_Net/transaction.vb
Imports System
Imports System.Data

Namespace DotNetDataProviderTemplate

    Public Class TemplateTransaction
        Implements IDbTransaction

        Public ReadOnly Property IsolationLevel() As IsolationLevel Implements IDbTransaction.IsolationLevel
            '*
            '* Should return the current transaction isolation
            '* level. For the template, assume the default
            '* which is ReadCommitted.
            '*
            Get
                Return IsolationLevel.ReadCommitted
            End Get
        End Property

        Public Sub Commit() Implements IDbTransaction.Commit
            '*
            '* Implement Commit here. Although the template does
            '* not provide an implementation, it should never be 
            '* a no-op because data corruption could result.
            '*
        End Sub

        Public Sub Rollback() Implements IDbTransaction.Rollback
            '*
            '* Implement Rollback here. Although the template does
            '* not provide an implementation, it should never be
            '* a no-op because data corruption could result.
            '*
        End Sub

        Public ReadOnly Property Connection() As IDbConnection Implements IDbTransaction.Connection
            '
            ' Set and retrieve the connection for the current transaction.
            '

            Get
                Return Nothing
            End Get
        End Property

        Public Sub Dispose() Implements IDisposable.Dispose
            Me.Dispose(True)
            System.GC.SuppressFinalize(Me)
        End Sub

        Private Sub Dispose(ByVal disposing As Boolean)
            If disposing Then
                If (Not Me.Connection Is Nothing) Then
                    ' implicitly rollback if transaction still valid
                    Me.Rollback()
                End If
            End If
        End Sub

    End Class
End Namespace

File: src/NBdbm/CS/Classes/Fachada_Net/transaction.cs
using System.Data;
using System.Collections;
using Microsoft.VisualBasic;
using System.Diagnostics;
using System;


namespace NBdbm
{
	namespace DotNetDataProviderTemplate
	{
		
		public class TemplateTransaction : IDbTransaction
		{
			
			
			public IsolationLevel IsolationLevel
			{
				//*
				//* Should return the current transaction isolation
				//* level. For the template, assume the default
				//* which is ReadCommitted.
				//*
				get
				{
					return IsolationLevel.ReadCommitted;
				}
			}
			
			public void Commit()
			{
				//*
				//* Implement Commit here. Although the template does
				//* not provide an implementation, it should never be
				//* a no-op because data corruption could result.
				//*
			}
			
			public void Rollback()
			{
				//*
				//* Implement Rollback here. Although the template does
				//* not provide an implementation, it should never be
				//* a no-op because data corruption could result.
				//*
			}
			
			public IDbConnection Connection
			{
				//
				// Set and retrieve the connection for the current transaction.
				//
				
				get
				{
					return null;
				}
			}
			
			public void Dispose()
			{
				this.Dispose(true);
				System.GC.SuppressFinalize(this);
			}
			
			private void Dispose(bool disposing)
			{
				if (disposing)
				{
					if (this.Connection != null)
					{
						// implicitly rollback if transaction still valid
						this.Rollback();
					}
				}
			}
			
		}
	}
	
}
